backend/handlers/setup/setuphandler.go

```diff
@@ -6,17 +6,18 @@ import (
 	"socialpredict/setup"
 )
 
-func GetSetupHandler(w http.ResponseWriter, r *http.Request) {
+func GetSetupHandler(loadEconomicsConfig func() (*setup.EconomicConfig, error)) func(w http.ResponseWriter, r *http.Request) {
```

Comment:
Hey I just had to add in a comment here and say...great idea. I did not consider this type of function signature, which helps separate concerns. Great idea and now I'm wondering where else in the code base this pattern might be applied. Thanks much.

Reply:
There is a negligible overhead to this in terms of performance, but dependency injection is the recommended way to make functions more testable.